### PR TITLE
[dynamo] Remove hardcoded CUDA references in test_logging.py

### PR DESCRIPTION
**Pull request by RiyaP2508**
## Summary

Makes `test/dynamo/test_logging.py` device-agnostic by fixing two tests that had hardcoded CUDA dependencies:

- **`test_ddp_graphs`**: Replaced `@requires_cuda_and_triton` with `@requires_gpu` and substituted hardcoded `"cuda:0"` with the module-level `device_type` variable. Made `device_ids` conditional since it is a CUDA-specific DDP parameter.
- **`test_cudagraph_static_inputs`**: Added missing `@requires_cuda_and_triton` guard since the test relies on `mode="reduce-overhead"` (CUDA graphs). Placed tensor on the appropriate device and widened the log record upper bound to accommodate additional CUDA-path logging.

## Motivation

This is part of the effort to decouple the test suite from specific hardware, enabling new accelerator backends to reuse existing tests without modification.

## Test plan

- `TEST_CONFIG=cpu python3 test/run_test.py -i dynamo/test_logging` — all tests pass
- `test_ddp_graphs` passes on CUDA
- `test_cudagraph_static_inputs` passes on CUDA with updated bound

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @kadeng @chauhang @amjames @jataylo @azahed98

### CODE REVIEW

**Comment by jansel** [comment]
@claude explain those CI failures, check if they are related to this PR, and review the changes

**Review by jansel — APPROVED** [review]

File: test/dynamo/test_logging.py [hunk 1]
```diff
@@ -449,10 +449,14 @@ def forward(self, x):
         os.environ["MASTER_PORT"] = str(find_free_port())
         dist.init_process_group("gloo", rank=0, world_size=1)
 
-        model = DDP(ToyModel().to("cuda:0"), device_ids=[0], bucket_cap_mb=4)
+        device = f"{device_type}:0"
```
Comment (fffrog) [inline test/dynamo/test_logging.py:452]: Hi @RiyaP2508, thank you very much for updates.

From my perspective, I think we should remove the global variable named `device_type` and try to convert part of those tests into `instantiate_device_type_tests` form, which can benefit all the out of tree accelerator to reuse or skip those test smoothly.

**Review by fffrog — CHANGES_REQUESTED** [review]

File: test/dynamo/test_logging.py [hunk 2]
```diff
@@ -1327,17 +1331,18 @@ def f(x):
...
         def fn(x):
             return x + 1
 
-        x = torch.ones(2, 2)
+        x = torch.ones(2, 2, device=device_type)
         torch._dynamo.mark_static_address(x)
         fn(x)
         self.assertGreater(len(records), 0)
-        self.assertLess(len(records), 4)
+        self.assertLess(len(records), 8)
```
Comment (mansiag05) [inline test/dynamo/test_logging.py:1345]: Unsure, whta is causing the extra log records due to which we've to change the bound from 4 to 8. Could you please add this in the PR description.

**Review by mansiag05 — CHANGES_REQUESTED** [review]
Also, could you please add HardwareClassification label to the test classes?
